File: practice/fibonacci.cpp
#include <bits/stdc++.h>
using namespace std;

void fib(int n)
{
    int a = 0, b = 1;
    int x = n, ans = 0;
    while (n != 0)
    {
        // cout << a << " ";
        ans = a;
        int temp = a + b;
        a = b;
        b = temp;
        n--;
    }
    cout << x << "th term in the fibonacci series is: " << ans << endl;
}

int main()
{
    int n = 5;
    fib(n);
    return 0;
}

File: IPA preparation/15/Fibonacci.java
import java.util.*;

public class Fibonacci {

    public static void fibonacci(int n) {
        // if (n > 2) {
        // System.out.print(0 + " " + 1 + " ");
        // }
        System.out.print(0 + " ");
        int a = 0, b = 1;
        while (b < n) {
            int temp = a + b;
            System.out.print(b + " ");
            a = b;
            b = temp;
        }

    }

    public static void main(String[] args) {
        int n = 20;
        fibonacci(n);
    }
}
